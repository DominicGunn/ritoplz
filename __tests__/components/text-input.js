'use strict'

/* eslint-env jest */

import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import TextInput from '../../components/ui/text-input'

describe('ui text-input component', () => {
  it('should have props type text by default', () => {
    const wrapper = shallow(<TextInput />)
    expect(wrapper.find('input').props().type).toEqual('text')
  })

  it('should have props name undefined', () => {
    const wrapper = shallow(<TextInput />)
    expect(wrapper.find('input').props().name).toBeUndefined()
  })

  it('should have props label empty string', () => {
    const wrapper = shallow(<TextInput />)
    expect(wrapper.find('label').text()).toEqual('')
  })

  it('should have props placeholder undefined', () => {
    const wrapper = shallow(<TextInput />)
    expect(wrapper.find('input').props().placeholder).toBeUndefined()
  })

  it('should have props inputRef undefined', () => {
    const wrapper = shallow(<TextInput />)
    expect(wrapper.find('input').props().inputRef).toBeUndefined()
  })

  it('should have props inputValue undefined', () => {
    const wrapper = shallow(<TextInput />)
    expect(wrapper.find('input').props().inputValue).toBeUndefined()
  })

  it('should have props handleInputChange undefined', () => {
    const wrapper = shallow(<TextInput />)
    expect(wrapper.find('input').props().handleInputChange).toBeUndefined()
  })

  it('should have props name email', () => {
    const wrapper = shallow(<TextInput name="email" />)
    expect(wrapper.find('input').props().name).toEqual('email')
  })

  it('should have props label Email', () => {
    const wrapper = shallow(<TextInput label="Email" />)
    expect(wrapper.find('label').text()).toEqual('Email')
  })

  it('should have props placeholder Email address', () => {
    const wrapper = shallow(<TextInput placeholder="Email address" />)
    expect(wrapper.find('input').props().placeholder).toEqual('Email address')
  })

  describe('snapshots', () => {
    it('should render text input', () => {
      const tree = renderer.create(<TextInput />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('should render email input', () => {
      const tree = renderer.create(<TextInput type="email" />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('should render placeholder email', () => {
      const tree = renderer
        .create(<TextInput placeholder="email" type="email" />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
