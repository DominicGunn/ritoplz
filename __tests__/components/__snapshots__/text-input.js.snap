// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui text-input component snapshots should render email input 1`] = `
<div
  data-jsx={67896831}
>
  <label
    data-jsx={67896831}
  />
  <input
    data-jsx={67896831}
    name={undefined}
    onChange={[Function]}
    placeholder={undefined}
    type="email"
    value={undefined}
  />
  <span
    data-jsx={67896831}
  />
</div>
`;

exports[`ui text-input component snapshots should render placeholder email 1`] = `
<div
  data-jsx={67896831}
>
  <label
    data-jsx={67896831}
  />
  <input
    data-jsx={67896831}
    name={undefined}
    onChange={[Function]}
    placeholder="email"
    type="email"
    value={undefined}
  />
  <span
    data-jsx={67896831}
  />
</div>
`;

exports[`ui text-input component snapshots should render text input 1`] = `
<div
  data-jsx={67896831}
>
  <label
    data-jsx={67896831}
  />
  <input
    data-jsx={67896831}
    name={undefined}
    onChange={[Function]}
    placeholder={undefined}
    type="text"
    value={undefined}
  />
  <span
    data-jsx={67896831}
  />
</div>
`;
